README.md

```diff
@@ -394,10 +396,71 @@ The `extract` method takes no arguments:
     1. `v` must have the same type that `f` returns in `extend`.
 
 
+### Bifunctor
 
+A value that implements the Bifunctor specification must also implement
+the Functor specification.
 
+A value which satisfies the specification of a Bifunctor does not need to
+implement:
+
+* Functor's `map`; derivable as `f => this.bimap(a => a, f)`
+
+1. `p.bimap(a => a, b => b)` is equivalent to `p` (identity)
+2. `p.bimap(a => f(g(a)), b => h(i(b))` is equivalent to `p.bimap(g, i).bimap(f, h)` (composition)
+
+#### `bimap` method
+
+A value which has a Bifunctor must provide an `bimap` method. The `bimap`
+method takes two arguments:
+
+    c.bimap(f, g)
+
+1. `f` must be a function which returns a value
+
+    1. If `f` is not a function, the behaviour of `bimap` is unspecified.
+    2. `f` can return any value.
+
+2. `g` must be a function which returns a value
+
+    1. If `g` is not a function, the behaviour of `bimap` is unspecified.
+    2. `g` can return any value.
+
+3. `bimap` must return a value of the same Bifunctor.
+
+
+### Profunctor
+
+A value that implements the Profunctor specification must also implement
+the Functor specification.
+
+A value which satisfies the specification of a Profunctor does not need to
+implement:
+
+* Functor's `map`; derivable as `f => this.promap(a => a, f)`
```

Comment:
`this` works differently in arrow functions, we need to use `function` here.

Otherwise everything looks good to me now.